src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -76,8 +77,12 @@ public TaskData getTaskByID(long taskID) {
   }
 
   public void deleteTasks(ArrayList<Long> taskIDs) {
-    ArrayList<Key> keys = getKeysFromTaskIDs(taskIDs);
-    datastore.delete(keys);
+    if (taskIDs.isEmpty()) {
+      return;
+    }
+    datastore.delete(getKeysFromTaskIDs(taskIDs));
+    Query query = new Query("Task").addFilter("parentTaskID", FilterOperator.IN, taskIDs);
+    deleteTasks(getTaskIDsFromTasks(getTasks(query, Integer.MAX_VALUE)));
```

Comment:
Is there a use case for limiting the number of records to pull?  

If you'd like to keep it, consider storing this on a better named constant like `NO_QUERY_LIMIT`, or maybe overload your method so you won't need to pass around MAX_VALUE if you need to pull all records.

Reply:
There's no use case now, but maybe there will be in the future (especially if we add a global search feature - to search through all tasks/projects visible to a user).